libraries/AP_Scripting/drivers/SIM7600.lua

```diff
@@ -0,0 +1,367 @@
+
+--[[
+    // @Param: LTE_SERVER_IP2
+    // @DisplayName: Server IP 2
+    // @Description: Third octet of the server IP address to connect to
+    // @Range: 0 255
+    // @User: Standard
+--]]
+local LTE_SERVER_IP2  = bind_add_param('SERVER_IP2',  6, 83)
+
+--[[
+    // @Param: LTE_SERVER_IP3
+    // @DisplayName: Server IP 3
+    // @Description: Fourth octet of the server IP address to connect to
+    // @Range: 0 255
+    // @User: Standard
+--]]
+local LTE_SERVER_IP3  = bind_add_param('SERVER_IP3',  7, 174)
+
+--[[
+    // @Param: LTE_SERVER_PORT
+    // @DisplayName: Server Port
+    // @Description: IPv4 Port of the server to connect to
+    // @Range: 1 65525
+    // @User: Standard
+--]]
+local LTE_SERVER_PORT = bind_add_param('SERVER_PORT',  8, 0)
+
+--[[
+    // @Param: LTE_BAUD
+    // @DisplayName: Serial Baud Rate
+    // @Description: Baud rate for the serial port to the LTE modem. If using something other than 115200 you need to connect to the modem and use AT+IPREX=BAUD to set the baud rate and save with AT&W
+    // @Values: 19200:19200,38400:38400,57600:57600,115200:115200,230400:230400,460800:460800,921600:921600
+    // @User: Standard
+--]]
+local LTE_BAUD        = bind_add_param('BAUD',  9, 115200)
+
+--[[
+    // @Param: LTE_TIMEOUT
+    // @DisplayName: Timeout
+    // @Description: Timeout in seconds for the LTE connection. If no data is received for this time, the connection will be reset.
+    // @Range: 1 60
+    // @Units: s
+    // @User: Standard
+--]]
+local LTE_TIMEOUT     = bind_add_param('TIMEOUT', 10, 10)
+
+local uart = serial:find_serial(LTE_SERPORT:get())
+if not uart then
+    gcs:send_text(MAV_SEVERITY.ERROR, 'SIM7600: could not find serial port')
+    return
+end
+
+local MAVLINK2 = 2
+
+local ser_device = serial:find_simulated_device(MAVLINK2, LTE_SCRPORT:get())
+if not ser_device then
+    gcs:send_text(MAV_SEVERITY.ERROR, 'SIM7600: could not find simulated device')
+    return
+end
+
+--[[
+    steps to connect the modem:
+    1. send "ATI" to get modem info and confirm communication
+    2. send AT+CIPMODE=1 to set to transparent mode
+    3. send AT+NETOPEN to open the network stack
+    4. send AT+CIPOPEN=0,"TCP","<server_ip>",<server_port> to open a TCP connection
+
+    Once connected we have a transparent connection to the server.
+    If we see "\r\nCLOSED\r\n" we need to reconnect.
+--]]
+
+local step = "ATI"
+
+uart:begin(LTE_BAUD:get())
+
+--[[
+    Open a log file to log the output from the modem
+    This is useful for debugging the connection process
+--]]
+local log_file = io.open('SIM7600.log', 'w')
+
+--[[
+    Function to read from the UART and log the output
+    This function reads up to 512 bytes at a time and writes it to the log file
+    returns the string read or nil
+--]]
+local function uart_read()
+    local s = uart:readstring(512)
+    if s and #s > 0 and log_file then
+        log_file:write('[' .. s .. ']\n')
+        log_file:flush()
+    end
```

Comment:
Should this be on by default? A bit concerned about overhead.

Reply:
it only logs during the connection phase, not once connected, and is very useful for support
the file ends up quite small